sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -54,6 +55,32 @@ void __iomem *get_mbox_reg_base(u32 id)
 }
 EXPORT_SYMBOL(get_mbox_reg_base);
 
+static int mt8195_get_mailbox_offset(struct snd_sof_dev *sdev)
+{
+	return MBOX_OFFSET;
+}
+
+static int mt8195_get_window_offset(struct snd_sof_dev *sdev, u32 id)
+{
+	return MBOX_OFFSET;
+}
+
+static int mt8195_ipi_init(struct snd_sof_dev *sdev)
+{
+	return mt8195_mbox_init(sdev);
+}
+
+static int mt8195_send_msg(struct snd_sof_dev *sdev,
+			   struct snd_sof_ipc_msg *msg)
+{
+	sof_mailbox_write(sdev, sdev->host_box.offset, msg->msg_data,
+			  msg->msg_size);
+
+	adsp_ipi_send(sdev, ADSP_IPI_MBOX_REQ, ADSP_IPI_OP_REQ);
```

Comment:
this can file, so ideally you should return the error to indicate if the msg TX failed